master/buildbot/scripts/upgrade_master.py

```diff
@@ -106,34 +107,38 @@ def upgradeDatabase(config, master_cfg):
 
 
 @in_reactor
-@defer.inlineCallbacks
 def upgradeMaster(config, _noMonkey=False):
     if not _noMonkey:  # pragma: no cover
         monkeypatches.patch_all()
 
     if not base.checkBasedir(config):
-        defer.returnValue(1)
-        return
+        return defer.succeed(1)
```

Comment:
AFAIK `defer.returnValue` raises exception, so no, `return` will not be executed. But in this way it's obvious even for non-Twisted developer (or static analysis tool), that function will return here, so I don't this this is a problem.

@tardyp ah, sorry, I missed that you *removed* inlineCallbacks, it's all OK!